presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -0,0 +1,427 @@
+                viewDefinition.ifPresent(definition ->
+                        addRecord(
+                                schemaPrefix.getCatalogName(),
+                                schemaPrefix.getSchemaName().get(),
+                                tableName,
+                                definition.getOriginalSql()));
+            }
+            return;
+        }
+        for (Map.Entry<QualifiedObjectName, ConnectorViewDefinition> entry : metadata.getViews(session, schemaPrefix).entrySet()) {
+            addRecord(
+                    entry.getKey().getCatalogName(),
+                    entry.getKey().getSchemaName(),
+                    entry.getKey().getObjectName(),
+                    entry.getValue().getOriginalSql());
+        }
+    }
+
+    private void addRecord(Object... values)
+    {
+        rawPageBuilder.declarePosition();
+        for (int i = 0; i < columnTypes.size(); i++) {
+            writeNativeValue(columnTypes.get(i), rawPageBuilder.getBlockBuilder(i), values[i]);
+        }
+
+        if (rawPageBuilder.isFull()) {
+            flushPageBuilder();
+        }
+    }
+
+    private void flushPageBuilder()
+    {
+        if (!rawPageBuilder.isEmpty()) {
+            rawPages.add(rawPageBuilder.build());
+            memoryUsageBytes += rawPageBuilder.getRetainedSizeInBytes();
+            rawPageBuilder.reset();
+        }
+    }
+
+    private static InformationSchemaTableType getTableTypeBySchemaTableName(SchemaTableName schemaTableName)
+    {
+        if (schemaTableName.equals(InformationSchemaMetadata.TABLE_COLUMNS)) {
+            return InformationSchemaTableType.TABLE_COLUMNS;
+        }
+        if (schemaTableName.equals(InformationSchemaMetadata.TABLE_TABLES)) {
+            return InformationSchemaTableType.TABLE_TABLES;
+        }
+        if (schemaTableName.equals(InformationSchemaMetadata.TABLE_VIEWS)) {
+            return InformationSchemaTableType.TABLE_VIEWS;
+        }
+        if (schemaTableName.equals(InformationSchemaMetadata.TABLE_SCHEMATA)) {
+            return InformationSchemaTableType.TABLE_SCHEMATA;
+        }
+        if (schemaTableName.equals(InformationSchemaMetadata.TABLE_TABLE_PRIVILEGES)) {
+            return InformationSchemaTableType.TABLE_TABLE_PRIVILEGES;
+        }
+        if (schemaTableName.equals(InformationSchemaMetadata.TABLE_ROLES)) {
+            return InformationSchemaTableType.TABLE_ROLES;
+        }
+        if (schemaTableName.equals(InformationSchemaMetadata.TABLE_APPLICABLE_ROLES)) {
+            return InformationSchemaTableType.TABLE_APPLICABLE_ROLES;
+        }
+        if (schemaTableName.equals(InformationSchemaMetadata.TABLE_ENABLED_ROLES)) {
+            return InformationSchemaTableType.TABLE_ENABLED_ROLES;
+        }
+        else {
+            throw new IllegalArgumentException(format("table does not exist: %s", schemaTableName));
```

Comment:
nit: redundant `else`; no need to use `format` (we don't use `format` when %s is the last thing in the format string)